### PR TITLE
test: fetch TeamMembers

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

[AR-1220](https://wearezeta.atlassian.net/browse/AR-1220)

Added tests for:
- TeamRepositoryTest
- And added a new test file: `UserMapperTest`, which it is only testing the TeamMember to UserEntity mapping function.

### Dependencies (Optional)

Needs releases with:

- [X] #276 
- [X] #279
- [X] #284

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt [hunk 1]
```diff
@@ -103,4 +107,72 @@ class TeamRepositoryTest {
...
+
+    @Test
+    fun givenTeamIdAndUserDomain_whenFetchingTeamMembers_thenTeamMembersShouldBeSuccessful() = runTest {
+        val teamMember = TeamsApi.TeamMember(
+            nonQualifiedUserId = "teamMember1",
+            createdAt = "01011970",
+            legalHoldStatus = LegalHoldStatusResponse.NO_CONSENT,
+            createdBy = "nonQualiefiedUserId1",
+            permissions = TeamsApi.Permissions(copy = 1, own = 1)
+        )
```
Comment (yamilmedina) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:119]: Would it be a good idea to extract this duplicated stub to an external class/function, so we can reuse it ?

**Review by yamilmedina — COMMENTED** [review]
Just a minor suggestion about a possible component we could reuse for testing, but apart from that, everything looks good! =)

**Review by typfel — APPROVED** [review]

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt (hunk 1 above)
Comment (alexandreferris) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:119]: I was thinking about something like having another module called: test-utils and only have [Data]Generator.kt classes, where we could generate/create any type of classes we need

Comment (vitorhugods) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:119]: @alexandreferris `test-utils` would depend on `network` to know what `TeamsApi.TeamMember` is. And `network` wouldn't be able to import `test-utils`. Only `logic` would be able to import it.

So it's as good as just extracting to another file right now.
At least we reduce redundancy inside `logic`.

Comment (alexandreferris) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt]: Great point @vitorhugods !

I guess then we will only be able to move forward on having the `test-utils` once we also have a specific domain model classes module.

For now I have added `TeamGenerator.kt` to create both a TeamDTO and a TeamMember (based on the scope of this ticket).

**Review by yamilmedina — APPROVED** [review]
looks great! 🚀  🌔

**Pull request merged**